dsa/src/size.rs

```diff
@@ -0,0 +1,32 @@
+/// DSA key size
+pub struct KeySize {
+    /// Bit size of p
+    pub(crate) l: u32,
+
+    /// Bit size of q
+    pub(crate) n: u32,
+}
+
+impl KeySize {
+    /// DSA parameter size constant: L = 1024, N = 160
+    #[deprecated(
+        note = "This size constant has a security strength of under 112 bits per SP 800-57 Part 1 Rev. 5"
+    )]
+    pub const DSA_1024_160: Self = Self { l: 1024, n: 160 };
+
+    /// DSA parameter size constant: L = 2048, N = 224
+    pub const DSA_2048_224: Self = Self { l: 2048, n: 224 };
+
+    /// DSA parameter size constant: L = 2048, N = 256
+    pub const DSA_2048_256: Self = Self { l: 2048, n: 256 };
+
+    /// DSA parameter size constant: L = 3072, N = 256
+    pub const DSA_3072_256: Self = Self { l: 3072, n: 256 };
+
+    /// Define a custom parameter size
+    ///
+    /// **⚠ YOU MOST LIKELY DO NOT NEED TO AND SHOULD NOT USE THIS!**
+    pub const fn custom(l: u32, n: u32) -> Self {
+        Self { l, n }
+    }
```

Comment:
One nit/sidebar looking at this:

It doesn't maintain any kind of invariants, which makes it somewhat pointless unless the goal is to potentially add invariants in the future.

Invariants could be maintained simply by using `assert!`, which will const panic at compile time.

I'd suggest either adding some invariants, or making `l` and `n` into `pub` fields so they can be set directly.

Reply:
That was actually one of the things I wanted to talk about.  

Does it make sense to keep a functionality for defining custom sizes or should we just say "NIST recommends these sizes. Choose one of them"?  

Because those the non-standard parameters could still be generated by an external tool and then imported and used with the `dsa` crate to generate key pairs if having non-standard parameter sizes is that important

Reply:
We could remove it for now and see if a feature request for it comes up